src/main/kotlin/app/UserService.kt

```diff
@@ -29,41 +26,13 @@ object UserService {
         val remainingRequests by lazy { GhService.remainingRequests }
         val hasFreeRemainingRequests by lazy { remainingRequests > (freeRequestCutoff ?: remainingRequests) }
         return Config.unrestricted()
-                || Cache.contains(user)
+                || (CacheService.lookUpInCache(user) != null)
```

Comment:
since this lookup operation is more expensive now, may be good to cache it `by lazy`

Reply:
I'm not sure if I got you right 😕:
[changes](https://github.com/tipsy/profile-summary-for-github/pull/113/commits/07071889e2d3488200746a314bc6efed83c5c982)

Reply:
actually I didn't read the diff properly; for `canLoadUser`, no `by lazy` is needed (the later use isn't pulled from the variable) so what you had originally is fine

ideally we would also create a `getUserIfCanLoad` method that would hold onto the cached profile so the db isn't hit twice (but this can be done in a later PR)

Reply:
> ideally we would also create a `getUserIfCanLoad` method that would hold onto the cached profile so the db isn't hit twice (but this can be done in a later PR)

Sounds somewhat challenging but let me give a try.

Reply:
Hope this is close to what you meant: [`1d04b35`](https://github.com/tipsy/profile-summary-for-github/pull/113/commits/1d04b350a914585197fd3103acb0fdd8c40e9567)